io-uring-test/src/tests/futex.rs

```diff
@@ -0,0 +1,182 @@
+use crate::Test;
+use io_uring::types::FutexWaitV;
+use io_uring::{cqueue, opcode, squeue, IoUring};
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+use std::{io, ptr, thread};
+
+// Not defined by libc.
+//
+// From: https://github.com/torvalds/linux/blob/v6.7/include/uapi/linux/futex.h#L63
+const FUTEX2_SIZE_U32: u32 = 2;
+
+const USER_DATA: u64 = 0xDEAD_BEEF_DEAD_BEEF;
```

Comment:
I suggestion using different user_data for each entry to avoid an entry from one test is returned in next.

Reply:
I'm expecting the test harness construct individual io-uring for each test, but apparently it does not. :thinking:  Fixed now.